runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/config/MqttKafkaConditionMatcher.java

```diff
@@ -14,42 +14,81 @@
  */
 package io.aklivity.zilla.runtime.binding.mqtt.kafka.internal.config;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 import io.aklivity.zilla.runtime.binding.mqtt.kafka.config.MqttKafkaConditionConfig;
 
 public class MqttKafkaConditionMatcher
 {
-    private final Matcher topicMatcher;
-    public final String topic;
+    private final List<Matcher> subscribeMatchers;
+    private final List<Matcher> publishMatchers;
+    public final MqttKafkaConditionKind kind;
 
     public MqttKafkaConditionMatcher(
         MqttKafkaConditionConfig condition)
     {
-        this.topic = condition.topic;
-        this.topicMatcher = condition.topic != null ? asTopicMatcher(condition.topic) : null;
+        this.subscribeMatchers =
```

Comment:
Given that `MqttKafkaConditionConfig` can only be `subscribe` or `publish`, then we need `kind` on `MqttKafkaConditionConfig` and a generic `topics` field, instead of separate storage for `subscribes` and `publishes` on `MqttKafkaConditionConfig`, but without exposing `kind` as an explicit property in the JSON schema patch.

Then the kind can propagate here